potential kernel arguments options support for all installers (HMS-8999)

mvo5 opened the pull request:
When debugging a test failure of centos9 iso install I want to use the cool new feature by Simon to add kernel arguments to the installer (c.f. https://github.com/osbuild/images/pull/1727/files) test in bib via a blueprint kernel append.

I noticed then that its currently implemented for a subset of our installers, so I created this quick draft to see what it would take and it seems straightforward. Is there a reason not to support the kargs for the missing installers? If not I'm happy to flesh this one out more, it needs tests (obviously) but I did not want to write them before there is agreement that is what we want.

[edit/reminder: the AanacondaContainerInstaller needs a coresponding change in bib to fully work]

Note that while this enables kernel.append in the lower levels there is a blueprint options check that will prevent this for e.g. edge-installer so if we want to support this for real we will need to enable it in the `checkOptions()` for blueprints code (which is currently not fine-grained enough to allow setting Kernel.append only but not kernel.name and we will need https://github.com/osbuild/images/pull/1216 first I think.

supakeen commented:
@mvo5 Why is this draft?

supakeen commented on pkg/image/ostree_simplified_installer.go:
```diff
@@ -139,7 +137,16 @@ func (img *OSTreeSimplifiedInstaller) InstantiateManifest(m *manifest.Manifest,
+	// the kernel options from the imageConfig but we only set
+	// those in the rawImage.OSTreeDeploymentCustomizations and
+	// not in the bootTreePipeline. Its unclear if we should change
+	// this or not. Its inconsistent with the other installers but
```
Yep, that's funky. Maybe this is from the time that we wanted blueprints to be able to apply to multiple image types even though silently discarding stuff is bad (in my opinion).

We can probably drop this from the allowed customizations as it has no effect?

achilleas-k replied:
The iot/edge-simplified-installer types create a raw image from an ostree commit and then put that on an ISO with the coreos installer.

The kernel arguments get applied to the raw image:
https://github.com/osbuild/images/blob/0a3a48f4bc8ec382d7e31f69535488cbb1e27fc2/pkg/distro/generic/images.go#L842

mvo5 replied:
I'm a Muppet (again) - I was confused by gen-manifests-diff not showing anything when doing https://github.com/osbuild/images/pull/1880 but its just a silly bug in gen-manifests-diff (I opened https://github.com/osbuild/images/pull/1881 to address that)

supakeen reviewed (approved):
Thank you, this is useful for at least all Anaconda installers as Anaconda switches behavior based on the kernel arguments present.

In particular this resolves the linked bug once we release the network installer.

mvo5 commented:
Sorry, conflicts again, force pushed again, this time from in checksum manifests (so trivial)

mvo5 commented:
And conflicts again :) But hopefully it calms down a bit now.

thozza reviewed (dismissed):
LGTM

achilleas-k commented:
> which is currently not fine-grained enough to allow setting Kernel.append only but not kernel.name and we will need #1216 first I think

I'm working on adding the `supported_blueprint_options` to RHEL right now.  Hope to have a PR up by tomorrow.

mvo5 commented:
> The option needs to be added to the image types that support it now. Let's also include it in one of the `test/configs/` for installers.

AFAICT we no longer allow kernel.append for any of our installers (via the new supported_options mechanism), even  https://github.com/osbuild/images/pull/1727/ seems to be no longer working. I'm happy to add it back but its orthogonal to this PR (this PR just enables the ability to add it and unifies the backend a bit). Should I open a new one that re-enables kernel.append for {minimal,image}-installer that also adds the config?

mvo5 commented:
[..]
> > [edit: I opened https://github.com/[/pull/1870](https://github.com/osbuild/images/pull/1870)]
> 
> Sorry, I got a bit confused and thought this PR was adding support for specifying kernel args in the blueprint for installer types.

No worries, I rebased now that #1870 is merged and this should be ready again (also fixed the typo that Thozza pointed out).
